jdaviz/core/template_mixin.py

```diff
@@ -20,15 +20,63 @@
 from jdaviz import __version__
 from jdaviz.core.events import (AddDataMessage, RemoveDataMessage,
                                 ViewerAddedMessage, ViewerRemovedMessage)
+from jdaviz.core.user_api import UserApiWrapper, PluginUserApi
 
 
-__all__ = ['TemplateMixin', 'PluginTemplateMixin',
-           'BasePluginComponent', 'BaseSelectPluginComponent',
+__all__ = ['show_widget', 'TemplateMixin', 'PluginTemplateMixin',
+           'BasePluginComponent', 'SelectPluginComponent',
            'SubsetSelect', 'SpatialSubsetSelectMixin', 'SpectralSubsetSelectMixin',
            'ViewerSelect', 'ViewerSelectMixin',
+           'LayerSelect', 'LayerSelectMixin',
            'DatasetSelect', 'DatasetSelectMixin',
            'AutoLabel', 'AutoLabelMixin',
-           'AddResults', 'AddResultsMixin']
+           'AddResults', 'AddResultsMixin',
+           'PlotOptionsSyncState']
+
+
+def show_widget(widget, loc, title):  # pragma: no cover
```

Comment:
Am I correct in interpreting this as giving us the ability to open _any_ individual widget in a new tab/sidecar, not just the top-level Application?

Reply:
technically, yes, although the _methods_ are only implemented at the helper and plugin-levels.

Reply:
You're now tempting me to implement something like `viz.show_viewer('spectrum-viewer', 'popout')`... (not in this PR though)